tests/unit/docs/test_utils.py

```diff
@@ -205,6 +206,86 @@ def test_hides_param_from_example(self):
         self.assert_not_contains_line(example)
 
 
+class TestHideUnusedShapeMember(BaseDocsTest):
+    def setUp(self):
+        super(TestHideUnusedShapeMember, self).setUp()
+        self.shape = "MyShape"
+        self.member = "MyMember"
+        self.hider = HideUnusedShapeMember(self.shape, self.member)
+
+    def test_hides_member_from_doc_string(self):
+        shape_section = self.doc_structure.add_new_section(self.shape)
+        shape_signature = ':param %s: ' % self.shape
+        shape_section.write(shape_signature)
+        member_section = shape_section.add_new_section(self.member)
+        member_signature = ':param %s: ' % self.member
+        member_section.write(member_signature)
+        self.assert_contains_line(shape_signature)
+        self.assert_contains_line(member_signature)
+        self.hider.hide_member(
+            'docs.response-params.s3.SampleOperation.complete-section',
+            self.doc_structure)
+        self.assert_contains_line(shape_signature)
+        self.assert_not_contains_line(member_signature)
+
+    def test_hides_member_from_example(self):
+        shape_section = self.doc_structure\
+            .add_new_section('structure-value')\
+            .add_new_section(self.shape)
+        shape_signature = ':%s: {' % self.shape
+        shape_section.write(shape_signature)
+        shape_section = shape_section\
+            .add_new_section('member-value')\
+            .add_new_section('structure-value')
+        member_section = shape_section.add_new_section(self.member)
+        member_signature = '    %s: \'string\'' % self.member
+        member_section.write(member_signature)
+        ending_section = shape_section.add_new_section('ending-bracket')
+        ending_section.write('}')
+        self.assert_contains_lines_in_order([
+            shape_signature, member_signature, '}'
+        ])
+        self.assert_contains_line(member_signature)
+        self.hider.hide_member(
+            'docs.response-example.s3.SampleOperation.complete-section',
+            self.doc_structure)
+        self.assert_contains_lines_in_order([
+            shape_signature, '}'
+        ])
+        self.assert_not_contains_line(member_signature)
+
+    def test_hides_member_from_example_and_removes_trailing_comma(self):
+        shape_section = self.doc_structure\
+            .add_new_section('structure-value')\
+            .add_new_section(self.shape)
+        shape_signature = ':%s: {' % self.shape
+        shape_section.write(shape_signature)
```

Comment:
This is pretty verbose. Pretty hard to imagine what the contents look like. Could we reuse one of the documenters or combine this into one helper method with comments.